### PR TITLE
[Capture] `iterative_qpe` uses qml control flow functions

### PR DESCRIPTION
**Pull request by albi3ro**
**Context:**

The capture project allows us to convert more complicated hybrid workflows into jaxpr.  

**Description of the Change:**

Makes it so that `iterative_qpe` can be compactly captured into jaxpr.

**Benefits:**

**Possible Drawbacks:**

The testing in this PR is horrible, but I don't really have a better way of doing it.  The produced jaxpr is rather complicated, the produced jaxpr is not compatible with `plxpr_to_tape`, we currently only use single-branch-statistics to simulate MCM's, and we can't post-process MCM's.

**Related GitHub Issues:**

[sc-72592]

### CODE REVIEW

File: pennylane/ops/functions/iterative_qpe.py [hunk 1]
```diff
@@ -28,12 +28,12 @@ def iterative_qpe(base, aux_wire, iters):
...
-      base (Operator): the phase estimation unitary, specified as an :class:`~.Operator`
-      aux_wire (Union[Wires, int, str]): the wire to be used for the estimation
-      iters (int): the number of measurements to be performed
+        base (Operator): the phase estimation unitary, specified as an :class:`~.Operator`
+        aux_wire (Union[Wires, int, str]): the wire to be used for the estimation
+        iters (int): the number of measurements to be performed
 
     Returns:
-      list[MidMeasureMP]: the list of measurements performed
+        list[MeasurementValue]: the abstract results of the mid circuit measurements
```
Comment (PietropaoloFrisoni) [inline pennylane/ops/functions/iterative_qpe.py:36]:
```suggestion
        list[MeasurementValue]: the abstract results of the mid-circuit measurements
```

File: tests/ops/functions/test_iterative_qpe.py [hunk 2]
```diff
@@ -237,3 +237,51 @@ def circuit_iterative():
...
+        qml.X(0)
+        return qml.iterative_qpe(qml.RZ(x, wires=[0]), aux_wire=1, iters=3)
+
+    x = jax.numpy.array(2.0)
+
+    jaxpr = jax.make_jaxpr(f)(1.5)
+
+    dev = qml.device("default.qubit", wires=3, seed=seed)
+
+    # hack for single branch statistics
```
Comment (PietropaoloFrisoni) [inline tests/ops/functions/test_iterative_qpe.py:272]:
```suggestion
    # hack for single-branch statistics
```

File: tests/ops/functions/test_iterative_qpe.py [hunk 3]
```diff
@@ -237,3 +237,51 @@ def circuit_iterative():
             return [qml.expval(op=i) for i in measurements]
 
         assert np.allclose(circuit_qpe(), circuit_iterative())
+
+
+@pytest.mark.slow
```
Comment (PietropaoloFrisoni) [inline tests/ops/functions/test_iterative_qpe.py:242]: My laptop takes around 1 minute and 20 seconds to run this, so I agree to mark it as slow

**Review by PietropaoloFrisoni — COMMENTED** [review]
Looks good!

File: tests/ops/functions/test_iterative_qpe.py (hunk 3 above)
Comment (albi3ro) [reply tests/ops/functions/test_iterative_qpe.py:242]: didn't actually know it was that bad 😢 . Really need to figure out a better way to test this.

File: tests/ops/functions/test_iterative_qpe.py [hunk 4]
```diff
@@ -237,3 +237,51 @@ def circuit_iterative():
...
+        return qml.iterative_qpe(qml.RZ(x, wires=[0]), aux_wire=1, iters=3)
+
+    x = jax.numpy.array(2.0)
+
+    jaxpr = jax.make_jaxpr(f)(1.5)
+
+    dev = qml.device("default.qubit", wires=3, seed=seed)
+
+    # hack for single branch statistics
+    samples = qml.math.vstack([dev.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, x) for _ in range(5000)])
```
Comment (mudit2812) [inline tests/ops/functions/test_iterative_qpe.py:273]: Oh man, hacky dynamic one shot 😬 

Unfortunately, I have absolutely no ideas about how else to test this other than verifying that the jaxpr equations are correct, which is not feasible.

**Review by PietropaoloFrisoni — APPROVED** [review]
I forgot to approve 🚀

**Pull request merged**